pkg/controller/dns/letsencrypt.go

```diff
@@ -0,0 +1,349 @@
+package dns
```

Comment:
Can you go through this file and ensure that only the things that are used outside of package have public visibility?

All the fields on the LEUser struct should be private and many of the functions and methods should also be private.

It'll help know exactly what we're exposing for consumption outside this package.

Reply:
Done.